p2p/host/peerstore/pstoremem/addr_book.go

```diff
@@ -46,7 +46,10 @@ type addrSegment struct {
 }
 
 func (segments *addrSegments) get(p peer.ID) *addrSegment {
-	return segments[byte(p[len(p)-1])]
+	if len(p) == 0 { // it's not terribly useful to use an empty peer ID, but at least we should not panic
```

Comment:
`p.Validate()` instead?

Reply:
How did that change sneak into this PR? That was https://github.com/libp2p/go-libp2p/pull/2006. Will rebase.